### ISSUE TITLE
reactstrap with react-router-bootstrap

### ISSUE THREAD

**Issue by FutureKode** [reactstrap/reactstrap#960]
I'm upgrading a header component from react-bootstrap to reactstrap, that's all fine, only thing is that the dropdown toggles are no longer active when a dropdown menu item is active. The previous code was like this:

```
<NavDropdown eventKey={3} title="Topology" id="topology-dropdown">
                <LinkContainer to="/topology-search">
                  <MenuItem>Search</MenuItem>
                </LinkContainer>
                <LinkContainer to="/model-query">
                  <MenuItem>Model</MenuItem>
                </LinkContainer>
              </NavDropdown>
```

The equivalent in `reactstrap`:

```
<UncontrolledDropdown nav inNavbar>
                <DropdownToggle nav caret>
                  Topology
                </DropdownToggle>
                <DropdownMenu>
                  <DropdownItem
                    tag={Link}
                    to="/topology-search"
                    activeClassName="active"
                  >
                    Search
                  </DropdownItem>
                  <DropdownItem
                    tag={Link}
                    to="/model-query"
                    activeClassName="active"
                  >
                    Model
                  </DropdownItem>
                </DropdownMenu>
              </UncontrolledDropdown>
```

Can I still use react-router-bootstrap with reactstrap?

**Comment by TheSharpieOne** [comment]
> Can I still use react-router-bootstrap with reactstrap?

No, react-router-bootstrap solves an issue for react-bootstrap which reactstrap solved with the `tag` prop.

As for the issue of "the dropdown toggles are no longer active when a dropdown menu item is active", it looks like that is a feature built into react-bootstrap which reactstrap is lacking, but bootstrap itself doesn't even have that feature (which is why reactstrap didn't implement it).

It can definitely be added via a prop toggle (like `inheritActive` or something) which would look to see if any of the children are active and add the active class. That way it would be optionally used and default to how bootstrap is.

**Comment by FutureKode** [comment]
thanks for this great library and helpful answer @TheSharpieOne 

I would like to contribute and take this on. I have forked the repo and will create a new branch according to the `contributing` docs in the readme. Should it be a patch version or a minor version?

**Comment by TheSharpieOne** [comment]
This would be an enhancement, which means when you create the commit, you'll use [conventional commits](https://conventionalcommits.org/#do-all-my-contributors-need-to-use-the-conventional-commit-specification) messages which starts with "feat(Dropdown): ". You don't need to bump the version or touch any files outside of the src and tests. (As long as you don't change the existing default behavior or existing features (as long as you are only adding new abilities), this would not be a breaking change.)
When it's merged and a release is created, we take all of the commits going into the release to determine if it is a major, minor, or patch.

**Comment by FutureKode** [comment]
Regarding the solution for this enhancement: looks like would have to array.find on props.children and it's children also, or is there a better way?

**Comment by TheSharpieOne** [comment]
you could iterating over the children (recessively since children will contain a menu which has the actual items) to determine if any of them are "active".
I suppose you could also use context to have the links communicate back up to inform the parent Dropdown that they are active/inactive... but that seems dirtier?

**Comment by FutureKode** [comment]
@TheSharpieOne thanks

Using this.context does sound dirtier especially as we would have to communicate that none of its children are active anymore.

**Comment by FutureKode** [comment]
We only want this for `UncontrolledDropdown` right?

**Comment by TheSharpieOne** [comment]
It's kinda half done. Right now when you have a child they is active the parent will be active with the inheritActive prop. But it will not automatically make the child active based on the route which is more similar what the other library is doing.
